concordium-contracts-common/src/schema_json.rs

```diff
@@ -1,47 +1,301 @@
+
+impl Display for JsonError {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            JsonError::FailedWriting => write!(f, "Failed writing"),
+            JsonError::UnsignedIntRequired => write!(f, "Unsigned integer required"),
+            JsonError::SignedIntRequired => write!(f, "Signed integer required"),
+            JsonError::FailedParsingAccountAddress => write!(f, "Failed parsing account address"),
+            JsonError::WrongJsonType(s) => write!(f, "{}", s),
+            JsonError::FieldError(s) => write!(f, "{}", s),
+            JsonError::EnumError(s) => write!(f, "{}", s),
+            JsonError::MapError(s) => write!(f, "{}", s),
+            JsonError::PairError(s) => write!(f, "{}", s),
+            JsonError::ArrayError(s) => write!(f, "{}", s),
+            JsonError::ParseError(s) => write!(f, "{}", s),
+            JsonError::ByteArrayError(s) => write!(f, "{}", s),
+            JsonError::FromHexError(e) => write!(f, "{}", e),
+            JsonError::TryFromIntError(e) => write!(f, "{}", e),
+            JsonError::ParseIntError(e) => write!(f, "{}", e),
+            JsonError::ParseDurationError(e) => write!(f, "{}", e),
+            JsonError::ParseTimestampError(e) => write!(f, "{}", e),
+            JsonError::TraceError {
+                ..
+            } => write!(f, "{}", self.print(false)),
+        }
+    }
+}
+
+impl TraceError for JsonError {
+    fn print_layer(&self, verbose: bool) -> (String, Option<&Self>) {
+        if let JsonError::TraceError {
+            error,
+            json,
+            field,
+        } = self
+        {
+            let formatted_json =
+                serde_json::to_string_pretty(json).unwrap_or_else(|_| format!("{}", json));
+            let message = if verbose {
+                format!("In {} of {}", field, formatted_json)
+            } else {
+                field.to_owned()
+            };
+            return (message, Some(error));
+        }
+
+        let message = format!("{}", self);
+        (message, None)
+    }
+
+    fn get_inner_error(&self) -> Option<&Self> {
+        if let JsonError::TraceError {
+            error,
+            ..
+        } = self
+        {
+            return Some(error);
+        }
+
+        None
+    }
+}
+
+impl JsonError {
+    /// Wraps a [JsonError] in a [JsonError::TraceError], providing a trace to
+    /// the origin of the error.
+    fn add_trace(self, field: String, json: &serde_json::Value) -> Self {
+        JsonError::TraceError {
+            field,
+            json: json.to_owned(),
+            error: Box::new(self),
+        }
+    }
+
+    /// Prints a formatted error message for variant. [JsonError::TraceError]
+    /// supports printing a verbose form including a more detailed
+    /// description of the error stack, which is returned if `verbose` is
+    /// set to true.
+    pub fn print(&self, verbose: bool) -> String { self.print_nested(verbose) }
```

Comment:
The method name here is also a bit misleading. If you change `print_nested`, then also change this one.